Mage.Sets/src/mage/cards/s/ShaileDeanOfRadiance.java

```diff
@@ -0,0 +1,115 @@
+package mage.cards.s;
+
+import mage.abilities.Ability;
+import mage.abilities.common.DiesCreatureTriggeredAbility;
+import mage.abilities.common.SimpleActivatedAbility;
+import mage.abilities.costs.common.TapSourceCost;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.DamageTargetEffect;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.counter.AddCountersTargetEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.VigilanceAbility;
+import mage.cards.CardSetInfo;
+import mage.cards.ModalDoubleFacesCard;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.counters.CounterType;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterControlledCreaturePermanent;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.game.Game;
+import mage.target.common.TargetCreaturePermanent;
+
+import java.util.UUID;
+
+/**
+ *
+ * @author htrajan
+ */
+public final class ShaileDeanOfRadiance extends ModalDoubleFacesCard {
+
+    private static final FilterCreaturePermanent filter = new FilterCreaturePermanent("another target creature");
+    private static final FilterPermanent embroseFilter = new FilterControlledCreaturePermanent("a creature you control with a +1/+1 counter on it");
+
+    static {
+        filter.add(AnotherPredicate.instance);
+        embroseFilter.add(CounterType.P1P1.getPredicate());
+    }
+
+    public ShaileDeanOfRadiance(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo,
+                new CardType[]{CardType.CREATURE}, new SubType[]{SubType.BIRD, SubType.CLERIC}, "{1}{W}",
+                "Embrose, Dean of Shadow", new CardType[]{CardType.CREATURE}, new SubType[]{SubType.HUMAN, SubType.WARLOCK}, "{2}{B}{B}");
+
+        // 1.
+        // Shaile, Dean of Radiance
+        // Legendary Creature - Bird Cleric
+        this.getLeftHalfCard().addSuperType(SuperType.LEGENDARY);
+        this.getLeftHalfCard().setPT(1, 1);
+
+        // Flying
+        this.getLeftHalfCard().addAbility(FlyingAbility.getInstance());
+
+        // Vigilance
+        this.getLeftHalfCard().addAbility(VigilanceAbility.getInstance());
+
+        // {T}: Put a +1/+1 counter on each creature that entered the battlefield under your control this turn.
+        this.getLeftHalfCard().addAbility(new SimpleActivatedAbility(new ShaileDeanOfRadianceEffect(), new TapSourceCost()));
```

Comment:
This can be simplified by creating a filter that checks if the creature entered this turn and then using `AddCountersAllEffect`. You also need to make sure that it only affects creatures that entered under the controller's control and not under another player's control but were stolen by the controller

Reply:
fixed